### PR TITLE
Get this into (near-)import shape.

### PR DESCRIPTION
**Pull request by ibara**
I think it's time this gets submitted to ports@ for further review and import.

### CODE REVIEW

**Comment by wmoxam** [comment]
Hey @ibara, this looks great. I'll test it out soon.

I'm wary of getting this specific version of Crystal approved for import, as it's a pre-release (https://github.com/crystal-lang/crystal/releases/tag/0.24.0) but it would be good to get a review.

Thanks!

**Comment by ibara** [comment]
I don't think it particularly matters, unless you think upstream will release another version of crystal also versioned 0.24.0, in that case I'd say wait.

File: lang/crystal/Makefile [hunk 1]
```diff
@@ -1,56 +1,76 @@
...
-	clang-5.0 -c -o ${WRKSRC}/src/llvm/ext/llvm_ext.o ${WRKSRC}/src/llvm/ext/llvm_ext.cc `llvm-config --cxxflags`
-	clang-5.0 -c -o ${WRKSRC}/src/ext/sigfault.o ${WRKSRC}/src/ext/sigfault.c
+	${CC} -c -o ${WRKSRC}/src/llvm/ext/llvm_ext.o \
+		${WRKSRC}/src/llvm/ext/llvm_ext.cc `llvm-config --cxxflags`
+	${CC} -c -o ${WRKSRC}/src/ext/sigfault.o ${WRKSRC}/src/ext/sigfault.c
 
-	mkdir ${WRKSRC}/.build
-	clang-5.0 ${WRKSRC}/../crystal.o -o ${WRKSRC}/.build/crystal -rdynamic  ${WRKSRC}/src/ext/sigfault.o ${WRKSRC}/src/llvm/ext/llvm_ext.o `(llvm-config --libs --system-libs --ldflags 2> /dev/null)` -lstdc++ -lpcre -lgc -lpthread -levent_core -levent_extra -lssl -liconv
+	mkdir -p ${WRKSRC}/.build
+	${CC} ${WRKSRC}/../crystal.o -o ${WRKBUILD}/crystal -rdynamic \
```
Comment (wmoxam) [inline lang/crystal/Makefile:52]: Needs to place binary at `${WRKBUILD}/.build/crystal` for build to succeed

**Review by wmoxam — CHANGES_REQUESTED** [review]
I tested and overall it works great. Just one small blocker found (perhaps missed because Crystal was already installed?)

**Pull request merged**